src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java

```diff
@@ -49,10 +63,24 @@ public MesosCloud(String name) throws InterruptedException, ExecutionException {
   public Collection<NodeProvisioner.PlannedNode> provision(Label label, int excessWorkload) {
     List<NodeProvisioner.PlannedNode> nodes = new ArrayList<>();
 
-    String slaveName = "undefined";
     while (excessWorkload > 0) {
-      nodes.add(new NodeProvisioner.PlannedNode(slaveName, startAgent(), 1));
-      excessWorkload--;
+      try {
+        LOGGER.log(
+            Level.INFO,
+            "Excess workload of "
+                + excessWorkload
+                + ", provisioning new Jenkins slave on Nomad cluster");
+        String slaveName = "undefined";
+
+        nodes.add(
+            new NodeProvisioner.PlannedNode(
+                slaveName,
+                MesosComputer.threadPoolForRemoting.submit(new ProvisioningCallback(this)),
```

Comment:
I don't think this has to be submitted on the thread. The interface is asynchronous already and all blocking calls such as `specMap.put` should be fast enough.

Reply:
resolved